server/protocols/webadmin/cli/james_cli

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env python3
+
+import click
+import json
+import requests as req
+import sys
+
+
+def print_json(obj):
+    parsed = json.loads(obj.text)
+    print(json.dumps(parsed, indent=4, sort_keys=True))
+
+
+@click.group()
+@click.option("--ip", default="127.0.0.1", show_default=True, help="Set ip address")
+@click.option("--port", default="8000", show_default=True, help="Set port")
+@click.pass_context
+def main(ctx, ip, port):
+    """Run './james_cli COMMAND --help' for more information on a command."""
+    path = "http://" + ip + ":" + port
+    ctx.obj = {
+        "path": path
+    }
+
+
+# Mapping routes implemantation ---------------------------------------
+@main.group()
+def mapping():
+    """Mapping routes implementation"""
+    pass
+
+
+@mapping.command(name="list")
+@click.pass_context
+def get_all_mapping(ctx):
+    """List all mappings"""
+    res = req.get(ctx.obj["path"] + "/mappings")
+    if res.status_code == 200:
+        print_json(res)
+    elif res.status_code == 401:
+        print("Authentication error.")
+    elif res.status_code == 500:
+        print("Internal server error!")
+    else:
+        print("Unhandled response received from the server.")
+        print("Status: " + str(res.status_code))
+        print_json(res)
+
+# Domain routes implemantation ---------------------------------------
+@main.group()
+def domain():
+    """Domain routes implementation"""
+    pass
+
+
+@domain.command(name="list")
+@click.pass_context
+def get_all_domains(ctx):
+    """Show all available domains"""
+    res = req.get(ctx.obj["path"] + "/domains")
+
+    if res.status_code == 200:
+        print("List of all available domains: ")
+        print_json(res)
+    elif res.status_code == 401:
+        print("Authentication error.")
+    elif res.status_code == 500:
+        print("Internal server error!")
+    else:
+        print("Unhandled response received from the server.")
+        print("Status: " + str(res.status_code))
+        print_json(res)
+
+
+@domain.command(name="add")
+@click.pass_context
+@click.argument("domain_name", type=str)
+def add_domain(ctx, domain_name):
+    """Add a domain to the domain list"""
+
+    res = req.put(ctx.obj["path"] + "/domains/" + domain_name)
+
+    if res.status_code == 204:
+        print("%s has been added." % domain_name)
+    elif res.status_code == 400:
+        print("Invalid request for domain creation.")
```

Comment:
Why?

Maybe we can return the json of the error returned by WebAdmin to help the user?

Reply:
how do it goes regarding compatibility with the previous CLI?

Reply:
I have checked on the manage-webadmin.md and code 400 turns out to be a "The domain name is invalid" message. The "Invalid request for domain creation" is what I copied from the swagger doc and it is not very clear.